_data-prepper/pipelines/configuration/processors/mutate-event.md

```diff
@@ -0,0 +1,304 @@
+
+```json
+{"message2": "goodbye"}
+```
+
+> If `message` does not exist in the event, then no action occurs.
+
+
+## RenameKey
+
+The `RenameKey` processor renames keys in an event.
+
+### Configuration
+
+You can configure the `RenameKey` processor with the following options.
+
+Option | Required | Description |
+| :--- | :--- | :--- |
+| `entries` | Yes | A list of entries to rename in an event. |
+| `from_key` | Yes | The key of the entry to be renamed. |
+| `to_key` | Yes | The new key of the entry. |
+| `overwrite_if_to_key_exists` | No | When set to `true`, the existing value is overwritten if `key` already exists in the event. The default value is `false`. |
+
+### Usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - rename_keys:
+        entries:
+        - from_key: "message"
+          to_key: "newMessage"
+          overwrite_if_to_key_exists: true
+  sink:
+    - stdout:
+```
+{% include copy.html %}
+
+
+Next, create a log file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with this filepath. For more information, see [Configuring Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/getting-started/#2-configuring-data-prepper).
+
+```json
+{"message": "hello"}
+```
+
+When you run the `RenameKey` processor, it parses the message into the following "newMessage" output:
+
+```json
+{"newMessage": "hello"}
+```
+
+> If `newMessage` already exists, its existing value is overwritten with `value`.
+
+
+
+### Special considerations
+
+Renaming operations occur in the order that the key value pair entries are listed in the `pipeline.yaml` file. This means that chaining (where key value pairs are renamed in sequence) is implicit with the `RenameKey` processor. See the following `pipline.yaml` file example:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - rename_key:
+        entries:
+        - from_key: "message"
+          to_key: "message2"
+        - from_key: "message2"
+          to_key: "message3"
+  sink:
+    - stdout:
+```
+
+Add the following contents to the `logs_json.log` file:
+
+```json
+{"message": "hello"}
+```
+{% include copy.html %}
+
+After the `RenameKey` processor runs, the following output appears:
+
+```json
+{"message3": "hello"}
+```
+
+## ConvertEntry
+
+The `ConvertEntry` processor converts a value type associated with the specified key in a message to the specified type. It is a casting processor that changes the types of some fields in the event or message. Some entered data needs to be converted to different types, such as an integer or a double, or a string to an integer, so that it will pass the events through condition-based processors or perform conditional routing. 
```

Comment:
Confirm that changing "key in a message" to "message key" retains the intended meaning.

Reply:
@chenqi0805 @asifsmohammed @dlvenable Can you please confirm?

Reply:
I would not use `message` but `event`.  These processors mutate `events` based on `keys`.  I would state:
```
The `ConvertEntry` processor converts a value type associated with the specified key in a event to the specified type. It is a casting processor that changes the types of some fields in events.
```